vsphere/resource_vsphere_compute_cluster.go

```diff
@@ -0,0 +1,1774 @@
+	meta interface{},
+	cluster *object.ClusterComputeResource,
+) error {
+	if !d.Get("force_evacuate_on_destroy").(bool) {
+		return nil
+	}
+
+	client, err := resourceVSphereComputeClusterClient(meta)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %s: Force-evacuating hosts in cluster before removal", resourceVSphereComputeClusterIDString(d))
+	hosts, err := resourceVSphereComputeClusterGetHostSystemObjects(
+		client,
+		structure.SliceInterfacesToStrings(d.Get("host_system_ids").(*schema.Set).List()),
+	)
+	if err != nil {
+		return err
+	}
+
+	if err := clustercomputeresource.MoveHostsOutOf(cluster, hosts, d.Get("host_cluster_exit_timeout").(int)); err != nil {
+		return fmt.Errorf("error force-removing old hosts out of cluster: %s", err)
+	}
+
+	return nil
+}
+
+// resourceVSphereComputeClusterApplyDelete process the removal of a
+// cluster.
+func resourceVSphereComputeClusterApplyDelete(d structure.ResourceIDStringer, cluster *object.ClusterComputeResource) error {
+	log.Printf("[DEBUG] %s: Proceeding with cluster deletion", resourceVSphereComputeClusterIDString(d))
+	if err := clustercomputeresource.Delete(cluster); err != nil {
+		return err
+	}
+	return nil
+}
+
+// resourceVSphereComputeClusterFlattenData saves the configuration attributes
+// from a ClusterComputeResource into the supplied ResourceData. It also saves
+// the root resource pool for the cluster in resource_pool_id.
+//
+// Note that other functions handle other non-configuration related items, such
+// as path, name, tags, and custom attributes.
+func resourceVSphereComputeClusterFlattenData(
+	d *schema.ResourceData,
+	meta interface{},
+	cluster *object.ClusterComputeResource,
+) error {
+	log.Printf("[DEBUG] %s: Saving cluster attributes", resourceVSphereComputeClusterIDString(d))
+	client, err := resourceVSphereComputeClusterClient(meta)
+	if err != nil {
+		return err
+	}
+
+	// Get the version of the vSphere connection to help determine what
+	// attributes we need to set
+	version := viapi.ParseVersionFromClient(client)
+
+	props, err := clustercomputeresource.Properties(cluster)
+	if err != nil {
+		return err
+	}
+
+	// Save the root resource pool ID so that it can be passed on to other
+	// resources without having to resort to the data source.
+	if err := d.Set("resource_pool_id", props.ResourcePool.Value); err != nil {
+		return err
+	}
+
+	return flattenClusterConfigSpecEx(d, props.ConfigurationEx.(*types.ClusterConfigInfoEx), version)
+}
+
+// expandClusterConfigSpecEx reads certain ResourceData keys and returns a
+// ClusterConfigSpecEx.
+func expandClusterConfigSpecEx(d *schema.ResourceData, version viapi.VSphereVersion) *types.ClusterConfigSpecEx {
+	obj := &types.ClusterConfigSpecEx{
+		DasConfig: expandClusterDasConfigInfo(d, version),
+		DpmConfig: expandClusterDpmConfigInfo(d),
+		DrsConfig: expandClusterDrsConfigInfo(d),
+	}
+
+	if version.Newer(viapi.VSphereVersion{Product: version.Product, Major: 6, Minor: 5}) {
```

Comment:
These are all doing the same version check. It looks like they either need to be checking different versions or can be combined into one conditional.